packages/client/src/generation/generateClient.ts

```diff
@@ -539,5 +545,18 @@ async function copyRuntimeFiles({ from, to, runtimeName, sourceMaps }: CopyRunti
     files.push(...files.filter((file) => file.endsWith('.js')).map((file) => `${file}.map`))
   }
 
-  await Promise.all(files.map((file) => fs.promises.copyFile(path.join(from, file), path.join(to, file))))
+  await Promise.all(files.map((file) => fs.copyFile(path.join(from, file), path.join(to, file))))
+}
+
+/**
+ * Attempts to delete the output directory.
+ * @param finalOutputDir
+ */
+async function deleteOutputDir(finalOutputDir: string) {
+  try {
+    // we want to make sure that if we delete, we delete the right directory
+    if (require(`${finalOutputDir}/package.json`).name === GENERATED_PACKAGE_NAME) {
```

Comment:
We only have a name in the package.json since [`4.2.0`](https://github.com/prisma/prisma/releases/tag/4.2.0) released 1 year ago
https://github.com/prisma/prisma/pull/14361

So what happens when someone upgrades from an older version?

Reply:
Then nothing happens, a bit of an edge case but also its better safety.